src/CModel.cc

```diff
@@ -806,11 +806,11 @@ class CModelStageImpl {
             result.likelihood->getUnweightedData()
         );
         data.amplitudes.deep() = lstsq.getSolution();
-        result.objective
-            = 0.5*(
-                result.likelihood->getUnweightedData().asEigen().cast<Scalar>()
-                - modelMatrix.asEigen().cast<Scalar>() * lstsq.getSolution().asEigen()
-            ).squaredNorm();
+        result.objective =
+                0.5 * (ndarray::asEigenMatrix(result.likelihood->getUnweightedData()).cast<Scalar>() -
+                       ndarray::asEigenMatrix(modelMatrix).cast<Scalar>() *
+                               ndarray::asEigenMatrix(data.amplitudes))
+                              .squaredNorm();
```

Comment:
It makes sense for the array whose squared norm is taken to be a variable but I'm not convinced it'll help to break it down further. A good name would be nice, but having no idea what the code is doing I'll use `tempArr` unless @TallJimbo wants to suggest something more descriptive.

Reply:
Call it `chi`, I think.

Reply:
@TallJimbo I assume I would have to save the result in a matrix (e.g. copy it). That copy is unnecessary in the original code. Is this a concern?
```
Eigen::MatrixXd chi =
        ndarray::asEigenMatrix(result.likelihood->getUnweightedData()).cast<Scalar>() -
        ndarray::asEigenMatrix(modelMatrix).cast<Scalar>() * ndarray::asEigenMatrix(data.amplitudes);
result.objective = 0.5 * chi.squaredNorm();
```

Reply:
For expediency I'm going to leave the code as it was. I agree it could use clarification but want to be sure to do it in a way that is safe and performant.